src/order/filter/basic.lean

```diff
@@ -2217,21 +2217,25 @@ lemma tendsto.not_tendsto {f : α → β} {a : filter α} {b₁ b₂ : filter β
   ¬ tendsto f a b₂ :=
 λ hf', (tendsto_inf.2 ⟨hf, hf'⟩).ne_bot.ne hb.eq_bot
 
-lemma tendsto_if {l₁ : filter α} {l₂ : filter β}
-    {f g : α → β} {p : α → Prop} [decidable_pred p]
-    (h₀ : tendsto f (l₁ ⊓ 𝓟 p) l₂)
-    (h₁ : tendsto g (l₁ ⊓ 𝓟 { x | ¬ p x }) l₂) :
+lemma tendsto.if {l₁ : filter α} {l₂ : filter β} {f g : α → β}
+  {p : α → Prop} {_ : ∀ x, decidable (p x)}
```

Comment:
I have a problem with this kind of implicit assumption vs. typeclass assumption. If you have a goal of the form `tendsto (λ x, if p x then f x else g x) l₁ l₂` then you can `apply tendsto.if` and your formulation is perfect because the implicit assumption will be picked from the goal. However, if you are trying to build a tendsto assertion using `h0.if h1` in a context where you have `h0` and `h1`, then if I understand correctly it won't work since there is nowhere Lean could extract the implicit argument from. Therefore, I think typeclass assumptions are better since they should work fine in both situations. What do you think?

Reply:
I copied implicit assumptions from the old code. Let me try to replace them with typeclass assumptions.

Reply:
I don't get it. In the old `tendsto_if`, the assumption was of the form `[decidable_pred p]`, so I thought you had switched to implicit on purpose.

Reply:
I think, I copied from the old `continuous_if'` assuming that it was there for a reason.

Reply:
And I messed up by using both versions in different lemmas.

Reply:
Pushed an update, waiting for CI.